### PR TITLE
DBZ-3792 Default value for Pravega scope

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-3792

### CODE REVIEW

**Comment by jpechane** [comment]
Failure caused by https://github.com/debezium/debezium/pull/2590

**Comment by gunnarmorling** [comment]
Hey @derekm, could you take a quick look at this change? Thx!

File: debezium-server/debezium-server-pravega/src/main/java/io/debezium/server/pravega/PravegaChangeConsumer.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
     @ConfigProperty(name = PROP_CONTROLLER, defaultValue = "tcp://localhost:9090")
     URI controllerUri;
 
-    @ConfigProperty(name = PROP_SCOPE)
+    @ConfigProperty(name = PROP_SCOPE, defaultValue = "debezium")
```
Comment (derekm) [inline debezium-server/debezium-server-pravega/src/main/java/io/debezium/server/pravega/PravegaChangeConsumer.java:49]: This seems like a reasonable default.

I don't know if Debezium has precedent for default group or namespace names elsewhere already. Maybe its Schema Registry integration or...?

**Review by derekm — APPROVED** [review]
I think this makes sense to do.

Sink docs say, "The Pravega sink expects destination scope and streams to already be created," but it feels like we could improve this by pointing to how destination names are formed, in terms of stream names. Not sure something like that necessarily belongs to this PR, though.

**Comment by derekm** [comment]
Oh, just looked at the jira issue, that's interesting, @jpechane! An un-activated sink's required parameter fails other sinks?

Could this also be solved at the CDI level by using [`Instance<T>`](https://docs.jboss.org/cdi/api/2.0/javax/enterprise/inject/Instance.html) to lazy-load (`select()`) the instance? It'll do the bean manager logic of `DebeziumServer` at least... But will it prevent MP Config from scanning all classes and enforcing required configs?

If you ever encounter a case of needing truly lazy requiredness, then I think you have to make it a runtime-resolved config instead of an annotation-resolved config. (That is, use `ConfigProvider.getConfig().getValue()` during construction.)

File: debezium-server/debezium-server-pravega/src/main/java/io/debezium/server/pravega/PravegaChangeConsumer.java (hunk 1 above)
Comment (gunnarmorling) [reply debezium-server/debezium-server-pravega/src/main/java/io/debezium/server/pravega/PravegaChangeConsumer.java:49]: I don't think there's precedent for this elsewhere, so this name is as good as any other.

**Comment by gunnarmorling** [comment]
> unless it is really necessary to do otherwise.

@jpechane, arguably, the case at hand is sort of problematic and would me make sway into that direction. Requiring a value for the configuration of another sink (and be it via mandating a default, as proposed here), seems just not alright in terms of isolation/encapsulation.

**Comment by gunnarmorling** [comment]
@derekm, on

> Sink docs say, "The Pravega sink expects destination scope and streams to already be created," but it feels like we could improve this by pointing to how destination names are formed, in terms of stream names.

Sure, any improvements are always welcomed. Can you send a PR with the change you have in mind? Doesn't need its own Jira issue, you could just reference this one here (DBZ-3792). Thx!

**Comment by gunnarmorling** [comment]
Thx for the update, @jpechane, I think that's better now. Thx for the quick feedback, @derekm!

**Pull request merged**